### PR TITLE
Rewrite document named property tests

### PR DESCRIPTION
**Pull request by ayg**
This incorporates whatwg/html#2930 and whatwg/html#2940.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17512)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/265626215)

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-17 15:40:14
Finished: 2017-08-17 15:50:54



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17512)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/265626215)

**Review by zcorpan — COMMENTED** [review]
Submitting minor nits from a quick review; I have not reviewed how well the tests match the spec.

File: html/dom/documents/dom-tree-accessors/named-item.html [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+    if (!documentExpected.has(key)) {
+      documentExpected.set(key, [val]);
+    } else {
+      documentExpected.get(key).push(val);
+    }
+  };
+
+  for (const elem of root.querySelectorAll("*")) {
+    // We don't use the IDL attributes in case the browser doesn't support the
+    // element at all (like applet at the time of writing).
```
Comment (ayg) [inline html/dom/documents/dom-tree-accessors/named-item.html]: No, but name is not on HTMLElement and does not work on applet if it's not supported.

File: html/dom/documents/dom-tree-accessors/named-item.html [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+      } else {
+        if (isArrayIndexPropertyName(id)) {
+          // WindowProxy has special behavior for these
+          // https://html.spec.whatwg.org/#windowproxy-getownproperty
+          windowUnexpected.add(id);
+        } else {
+          addToWindow(id, elem);
+        }
+        if (elem.localName == "applet" || elem.localName == "object" ||
+            (elem.localName == "img" && elem.name !== null &&
```
Comment (ayg) [inline html/dom/documents/dom-tree-accessors/named-item.html]: Good catch.  I assumed it worked the same as getAttribute() and returned null for an absent attribute.

**Comment by sideshowbarker** [comment]
w3c-test:mirror

**Comment by ayg** [comment]
I currently am unemployed, so I will not be able to work any further on this unless I succeed in finding someone who's willing to pay me to continue to work on web standards.

**Comment by annevk** [comment]
Rebased and replaced with #7266. Kept authorship intact.

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-19 19:49:11
Finished: 2017-08-19 19:52:02





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17570)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/266353384)

**Pull request closed without merging**